Branch-SDK/src/main/java/io/branch/referral/TrackingController.java

```diff
@@ -22,19 +23,32 @@ public class TrackingController {
     TrackingController(Context context) {
         updateTrackingState(context);
     }
-    
-    void disableTracking(Context context, boolean disableTracking) {
-        if (trackingDisabled != disableTracking) {
-            trackingDisabled = disableTracking;
-            if (disableTracking) {
-                onTrackingDisabled(context);
-            } else {
-                onTrackingEnabled();
+
+    void disableTracking(Context context, boolean disableTracking, @Nullable Branch.TrackingStateCallback callback) {
+        if (trackingDisabled == disableTracking) {
+            if (callback != null) {
+                callback.onTrackingStateChanged(trackingDisabled, Branch.getInstance().getFirstReferringParams(), null);
+            }
+            return;
+        }
+
+        trackingDisabled = disableTracking;
+        PrefHelper.getInstance(context).setBool(PrefHelper.KEY_TRACKING_STATE, disableTracking);
+
+        if (disableTracking) {
+            onTrackingDisabled(context);
+            if (callback != null) {
```

Comment:
Does tracking disabled also return a callback? If not I'd suggest adding, so that there's mirrored control for both enable and disable tracking.

Reply:
`onTrackingDisabled` does not have a callback since it just clears PrefHelper values. And `onTrackingEnabled` actually uses the BranchReferralInitListener callback, which it passes a few functions down to `ServerRequestRegisterOpen`. 

Does it still make sense to have a callback for `onTrackingDisabled`?

Reply:
My thinking is if tracking is explicitly disabled, the user should know when that completes. But if values are only getting set, then it should be all synchronous.

Reply:
Yeah thats what i'm thinking. It does all appear synchronous so this should technically return the TrackingStateCallback with the state when its done.